php/Blockonomics.php

```diff
@@ -752,45 +820,69 @@ public function check_callback_secret($secret){
         exit(__("Error: secret does not match", 'blockonomics-bitcoin-payments'));
     }
 
-    public function save_transaction($value, $order, $wc_order){
-        if (!metadata_exists('post',$wc_order->get_id(),'blockonomics_'. $order['crypto'] .'_txid', $order['txid']) )  {
-            update_post_meta($wc_order->get_id(), 'blockonomics_'. $order['crypto'] .'_txid', $order['txid']);
-            update_post_meta($wc_order->get_id(), 'expected_'. $order['crypto'], $order['satoshi']/1.0e8);
+    public function save_transaction($order, $wc_order){
+        $txid_meta_key = 'blockonomics_payments_txid';
+        $txid_meta_value = get_post_meta($order['order_id'], $txid_meta_key);
+        $txid = $order['txid'];
+        if (empty($txid_metavalue) || $txid_meta_value[0] == $txid){
+            update_post_meta($wc_order->get_id(), $txid_meta_key, $txid);
```

Comment:
We want the crypto to always be visible for all addresses, and not just for multiple payments:

![1st Address](https://user-images.githubusercontent.com/39666372/221190233-bd43737d-c8c9-475f-bd47-4aa09d6dba15.png)


The same for transaction id's. The crypto for each transaction should always be visible. Transaction Id's are also currently overwriting, instead of adding to the metadata:

![2nd Payment](https://user-images.githubusercontent.com/39666372/221190608-624a56f1-90fd-4a18-a3e9-15d002955541.png)

The final custom fields should look as follows:

![Full Display](https://user-images.githubusercontent.com/39666372/221190787-e07116fb-8cc9-4a4d-bc10-ea45d5c46d6c.png)

Reply:
I believe problem with having 1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa (btc) is that 1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa is not searchable in the Woocommerce filters because it matches the complete string. To force it is match part of thing requires complex logic.

Ultimately we want it in the format you suggest @DarrenWestwood
_````1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa (btc), qr3fmxznghk8h0af7mpj0ay590ev5js72chef5md3w(bch)`_

But for now lets go with a simpler format of
_1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa, qr3fmxznghk8h0af7mpj0ay590ev5js72chef5md3w_

This way for single address/transaction payments (which is 95% cases), search will still work. For multiple address/transactions search won't work we document as an issue now and fix later